dropshot_endpoint/src/metadata.rs

```diff
@@ -314,9 +328,72 @@ impl ValidatedEndpointMetadata {
 #[derive(Debug)]
 pub(crate) enum VersionRange {
     All,
-    From(semver::Version),
-    Until(semver::Version),
-    FromUntil(semver::Version, semver::Version),
+    From(VersionSpecifier),
+    Until(VersionSpecifier),
+    FromUntil(VersionSpecifier, VersionSpecifier),
+}
+
+impl syn::parse::Parse for VersionRange {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let lookahead = input.lookahead1();
+        if lookahead.peek(syn::Token![..]) {
+            let _ = input.parse::<syn::Token![..]>()?;
+            if input.is_empty() {
+                Ok(VersionRange::All)
+            } else {
+                let latest = input.parse::<VersionSpecifier>()?;
+                Ok(VersionRange::Until(latest))
+            }
+        } else {
+            let earliest = input.parse::<VersionSpecifier>()?;
+            let dotdot = input.parse::<syn::Token![..]>()?;
+            let lookahead = input.lookahead1();
+            if lookahead.peek(syn::LitStr) || lookahead.peek(syn::Ident) {
```

Comment:
Do you need to peek or can you if let Ok(..) parse?

Reply:
Good question!  I almost wrote that but I think you could ask the same thing about the existing code.  I inferred maybe a failed `parse()` might eat some tokens or otherwise leave the stream in the wrong state so I just went and did the lookahead here, even though this involves knowing what the `VersionSpecifier` accepts.  (We're also using `syn::Ident` instead of `syn::Path` because for reasons I don't understand `syn::Ident` has no data but `syn::Path` does so I wasn't sure how to construct that.)  Sorry this all feels janky but the jank at least feels contained (and correct).